Ortac

katrinafyi opened the pull request.

agle committed:
notes

katrinafyi committed:
ortac. TODO: wrapper

katrinafyi committed:
ocamlgraph first class module

katrinafyi committed:
blah. idk why it doesn't raise an error in this bad code.

idk if wrapper + qcheck is needed.

idk why using `t.size` in checks/requires is not translated
to the projection function

katrinafyi committed:
qcheck-stm works by adding Gen module. remove wrapper.

agle committed:
Merge branch 'main' into ortac

agle commented:
I want to merge this but not with the example test that doesn't test anything substantial, good candidates are

- the page table structure used by the interpreter
- the fresh identifier system, but its contract is ugly
- would value domain soundness make sense in ortac or just properties?

katrinafyi committed:
Merge remote-tracking branch 'origin/main' into ortac

Conflicts:
bincaml.opam
dune-project
lib/dune
lib/lang/common.ml
test/dune

katrinafyi committed:
restore opam file

katrinafyi committed:
ORTAC fdsjaiofjdsiaofdjsiaofjdsiaofdsjaio

    dune exec test/interp_spec_tests.exe -- --verbose

    File "../lib/lang/interp_spec.mli", line 12, characters 34-54:
    12 |   val write_bv : t -> addr:Z.t -> Util.Common.Bitvec.t -> unit
                                           ^^^^^^^^^^^^^^^^^^^^
    Error: Gospel error: No module with name Util.

katrinafyi commented:
superseded by https://github.com/agle/bincaml/pull/8 because ortac is too hard and opaque, so we should just use qcheck-stm directly.
